plugin/pkg/admission/persistentvolume/label/admission.go

```diff
@@ -111,6 +164,51 @@ func (l *persistentVolumeLabel) Admit(a admission.Attributes) (err error) {
 	return nil
 }
 
+func (l *persistentVolumeLabel) isPVCOkToDelete(pvc *api.PersistentVolumeClaim) bool {
+	// prevent deletion of a pvc that is bound to a pv
+	// if the pv is used by an active pod
+	// issue #45143
+
+	glog.Infof("IDC isPVCOkToDelete entered")
+
+	//check if pvc is bound to pv and associated pv name exists
+	if pvc != nil && pvc.Status.Phase == api.ClaimBound && pvc.Spec.VolumeName != "" {
+		glog.Infof("IDC isPVCOkToDelete pvc is bound")
+		//find the associated pv and check
+		volume, volumeErr := l.client.Core().PersistentVolumes().Get(pvc.Spec.VolumeName, metav1.GetOptions{})
+		if volumeErr == nil {
+			glog.Infof("IDC isPVCOkToDelete found bound volume %s", volume.Name)
+			//check all running and pending pods for this volume
+			pods, podErr := l.podLister.Pods(pvc.Namespace).List(labels.Everything())
+			if podErr == nil {
+				glog.Infof("IDC isPVCOkToDelete got pods")
+				for _, pod := range pods {
+					glog.Infof("IDC isPVCOkToDelete checking pod %q", pod.Name)
+					if pod.Status.Phase == api.PodRunning || pod.Status.Phase == api.PodPending {
```

Comment:
Asking @kubernetes/sig-storage-pr-reviews, do we want to allow deletion if the pod is terminated or failed? Or only if the pod object is deleted?

There is still a race condition between checking all the pods, and controllers destroying and creating pods.  I don't know if there's a good way we can prevent that.